src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,70 @@
+package nextstep.subway.application;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+	private LineRepository lineRepository;
+	private StationService stationService;
+
+	public LineService(
+		LineRepository lineRepository,
+		StationService stationService
+	) {
+		this.lineRepository = lineRepository;
+		this.stationService = stationService;
+	}
+
+	@Transactional
+	public LineResponse saveLine(LineRequest lineRequest) {
+		Line persistLine = lineRepository.save(new Line(
+			lineRequest.getName(),
+			lineRequest.getColor()
+		));
+
+		if(lineRequest.getUpStationId() != null && lineRequest.getDownStationId() != null) {
```

Comment:
```suggestion
		if (lineRequest.getUpStationId() != null && lineRequest.getDownStationId() != null) {
```
`request`에 대한 검증 좋습니다 👍 
다만 `request`에게 위임해도 좋을 것 같습니다 😄 

이번 미션은 사실 테스트가 더 중심이 되는지라 코드에 대한 의견은 많이 안드리려고 합니다만 🙏 
사실 두 값 모두 `null`이 아닐 때 처리하는게 맞습니다만 만약에 `null`이라면 여기까지 올 필요도 없을 거라 😅 
`validation`을 한번 활용해보시는 것도 좋을 것 같습니다 🙇 

도움 되실 만한 내용 공유 드립니다 🙇 
> https://meetup.toast.com/posts/223

Reply:
아! 이 부분은 어떻게 처리할지 좀 고민했던 부분인데요. 
미션 과제 설명에서 `프론트엔드-노선 페이지 화면`을 예로 보여준 부분에는
처음에 지하철 역을 입력하지 않고 노선 이름과 색상만 저장하더라구요.
그래서 부득이하게 검증하지 않고 null 체크만하고 저장하는 방식으로 진행했어요.

말씀하신대로 검증하는 방식으로 진행할까요? 😊

![image](https://user-images.githubusercontent.com/8567302/172089070-180fe166-5ba6-46bb-8697-ba159a2c2fc2.png)